com.vrcfury.vrcfury/Editor/VF/Service/OnDemandSyncService.cs

```diff
@@ -0,0 +1,111 @@
+                manager.GetParams().GetParam(param.Name()).networkSynced = false;
+            }
+            var pointer = fx.NewInt("SyncPointer", synced: true);
+            var localPointer = fx.NewInt("LocalPointer"); // Use a local only pointer to avoid race condition
+            var data = fx.NewFloat("SyncData", synced: true);
+            var sending = fx.NewFloat("IsSending");
+
+            var txLayer = fx.NewLayer("Send");
+            var txIdle = txLayer.NewState("Idle");
+            txIdle.Drives(sending, 0, true);
+            var txAdder = txLayer.NewState("Adder");
+            var txResetter = txLayer.NewState("Reset");
+            txResetter.Drives(localPointer, 0, local: true)
+                .TransitionsToExit()
+                .When(fx.Always());
+            txAdder.TransitionsTo(txResetter)
+                .When(localPointer.IsGreaterThan(onDemandSyncParameters.Count - 1)); // always > 0
+            txAdder.DrivesDelta(localPointer, 1)
+                .TransitionsToExit()
+                .When(fx.Always());
+            
+            // instant sync on change detection
+            for (int i = 0; i < onDemandSyncParameters.Count; i++) {
+                var src = onDemandSyncParameters[i];
+                var lastSrc = fx.NewFloat($"last{src.Name()}");
+                directTree.Add(math.MakeCopier(src, lastSrc)); 
+                var srcDiff = math.Subtract(src, lastSrc);
+                var maintain = math.MakeMaintainer(srcDiff); // Maintain while in send animation
+                directTree.Add(sending, maintain);
+                
+                var sendInstantState = txLayer.NewState($"Send Instant {onDemandSyncParameters[i]}");
+                sendInstantState
+                    .DrivesCopy(data, src)
+                    .Drives(pointer, i)
+                    .Drives(sending, 1, true)
+                    .TransitionsToExit()
+                    .WithTransitionExitTime(0.1f)
+                    .When(fx.Always());
+                txIdle.TransitionsTo(sendInstantState)
+                    .When(fx.IsLocal().IsTrue().And(srcDiff.IsGreaterThan(0).Or(srcDiff.IsLessThan(0))));
+            }
+
+            // round robin pointer iteration
+            for (int i = 0; i < onDemandSyncParameters.Count; i++) {
+                var src = onDemandSyncParameters[i];
+                var sendState = txLayer.NewState($"Send {onDemandSyncParameters[i]}");
+                sendState
+                    .DrivesCopy(data, src)
+                    .DrivesCopy(pointer, localPointer)
+                    .Drives(sending, 1, true)
+                    .TransitionsTo(txAdder)
+                    .WithTransitionExitTime(0.1f)
+                    .When(fx.Always());
+                txIdle.TransitionsTo(sendState)
+                    .When(fx.IsLocal().IsTrue().And(localPointer.IsEqualTo(i)));
+            }
+            
+            var rxLayer = fx.NewLayer("Receive");
```

Comment:
Same can apply for this, add an entry state first, make it go into a "local trap" for local, and transition to exit for non-local. That way you can leave off all the other islocal conditional checks below.